### PR TITLE
Init line lists

### PR DESCRIPTION
**Pull request by Kuroshii**
#### Changes
- Initialize point-indexed, distance sorted doubly-linked-list line structure for graph creation

---

### CODE REVIEW

File: ai_graph_color/problem_generator.py [hunk 1]
```diff
@@ -83,7 +87,69 @@ def build_graph(points):
...
+
+    :param points: the points to create lines from
+    :type points: list[tuple(float, float)]
+    :return: the map of lines
+    :rtype: map<frozenset(int), Line>
+    """
+    lines = {}
+    for point_a, point_b in combinations(enumerate(points), 2):
+        connecting_line = Line(point_a[1], point_b[1])
+        key = frozenset([point_a[0], point_b[0]])
```
Comment (sagersmith8) [inline ai_graph_color/problem_generator.py:120]: What is a `frozenset`? Also is it okay that it is immutable?

Comment (Kuroshii) [reply ai_graph_color/problem_generator.py:120]: [frozenset](https://docs.python.org/2/library/stdtypes.html#frozenset) is an immutable set

File: ai_graph_color/problem_generator.py [hunk 2]
```diff
@@ -83,7 +87,69 @@ def build_graph(points):
...
+    :param num_points: the number of points being connected
+    :type num_points: int
+    :return: the distance list
+    :rtype: list[dllist(tuple(int, Line))]
+    """
+    point_distances = [[] for _ in xrange(num_points)]
+    for point_pair, connecting_line in lines.iteritems():
+        if len(point_pair) != 2:
+            raise Exception(
+                ("Expected set indexes into set-line-map to have exactly 2 "
```
Comment (sagersmith8) [inline ai_graph_color/problem_generator.py:143]: Will this go to a logger or stdout? We want it to go to a `logger`

File: tests/test_problem_generator.py [hunk 3]
```diff
@@ -70,3 +70,72 @@ def test_scatter_points(self):
...
+        """
+        points = [(0.0, 0.0), (0.0, 3.0), (1.0, 1.0), (1.0, 5.0)]
+        lines = problem_generator.create_lines(points)
+        distance_list = problem_generator.create_distance_list(
+            lines, len(points)
+        )
+
+        for src_index, connections in enumerate(distance_list):
+            distances = map(lambda i: i[1].distance, connections)
+            self.assertTrue(all(
```
Comment (sagersmith8) [inline tests/test_problem_generator.py:115]: Nice use of the `all` command

File: tests/test_problem_generator.py [hunk 4]
```diff
@@ -70,3 +70,72 @@ def test_scatter_points(self):
...
+
+            self.assertAlmostEqual(distance, connecting_line.distance)
+            self.assertEqual(
+                frozenset(map(lambda i: points[i], pair)),
+                frozenset([connecting_line.left_point,
+                           connecting_line.right_point])
+            )
+            connecting_line.free()  # should not raise any errors
+
+    def test_create_distance_list(self):
```
Comment (sagersmith8) [inline tests/test_problem_generator.py:99]: Very verbose. I like it.

**Pull request merged**